Clicker/Extensions/Utils.swift

```diff
@@ -30,3 +32,80 @@ func decodeObjForKey(key: String) -> Any {
     let decodedData = UserDefaults.standard.value(forKey: key) as! Data
     return NSKeyedUnarchiver.unarchiveObject(with: decodedData)!
 }
+
+// MARK: - Utils for Polls
+func buildPollOptionsModel(from poll: Poll, userRole: UserRole) -> PollOptionsModel {
+    var type: PollOptionsModelType
+    switch poll.questionType {
+    case .freeResponse:
+        type = buildFROptionModelType(from: poll)
+    case .multipleChoice:
+        switch poll.state {
+        case .live, .ended:
+            switch userRole {
+            case .member:
+                type = buildMCChoiceModelType(from: poll)
+            case .admin:
+                type = buildMCResultModelType(from: poll)
+            }
+        case .shared:
+            type = buildMCResultModelType(from: poll)
+        }
+    }
+    return PollOptionsModel(type: type, pollState: poll.state)
+}
+
+func calculatePollOptionsCellHeight(for pollOptionsModel: PollOptionsModel, state: CardControllerState) -> CGFloat {
+    let verticalPadding: CGFloat = LayoutConstants.pollOptionsVerticalPadding * 2
+    var optionModels: [OptionModel]
+    var optionHeight: CGFloat
+    let isHorizontal = state == .horizontal
+    switch pollOptionsModel.type {
+    case .mcResult(resultModels: let mcResultModels):
```

Comment:
I'm pretty sure (not certain) you can just do `case .mcResult(let mcResultsModel)`